regression/cbmc/union10/main.c

```diff
@@ -0,0 +1,23 @@
+typedef struct
+{
+  union {
+    int access[256];
+  };
+} S1;
+
+typedef struct
+{
+  int c;
+} S2;
+
+static S1 (* const l[1]) = {((S1 *) ((0x50000000UL) + 0x00000) )};
+
+void main()
+{
+  __CPROVER_allocated_memory(0x50000000UL, sizeof(S1));
+  l[0]->access[0] = 0;
```

Comment:
Why is l a single-item array instead of a simple pointer? Are you trying to defeat optimisation or to exercise some particular behaviour? Either way add a comment clarifying what you're trying to do.